lib/neovim-plugin.nix

```diff
@@ -99,21 +102,36 @@ with lib;
             example = settingsExample;
           };
         }
+        // optionalAttrs allowLazyLoad {
+          lazyLoad = nixvimOptions.mkLazyLoadOption {
+            inherit originalName cfg;
+            optionsForPlugin = true;
+            lazyLoad = {
+              # TODO: extract extraConfigLua and extraConfigLuaPre from extraConfig
+              after = optionalString callSetup ''
+                require('${luaName}')${setup}(${optionalString (cfg ? settings) (toLuaObject cfg.settings)})
+              '';
+              colorscheme = if (isColorscheme && (colorscheme != null)) then [ colorscheme ] else null;
+            } // lazyLoad;
+          };
+        }
         // extraOptions;
 
       config =
         let
-          cfg = config.${namespace}.${name};
           extraConfigNamespace = if isColorscheme then "extraConfigLuaPre" else "extraConfigLua";
+          lazyLoaded = if (cfg ? lazyLoad) then cfg.lazyLoad.enable else false;
```

Comment:
This can just be:

```nix
lazyLoaded = cfg.lazyLoad.enable or false;
```

`or` defines a fallback for when an attribute doesn't exist.